### ISSUE TITLE
Pod ipBlock egress rule not applied when hostPort is defined

### ISSUE THREAD

**Issue by johnsler** [projectcalico/calico#5104]
I have a OKE cluster with calico enabled (installed with [calico.yaml](https://docs.projectcalico.org/manifests/calico.yaml)). I have a Pod and a NetworkPolicy allowing these labelled pods egress to a well known url http://169.254.169.254/opc/v2/instance/. So far so good. Now if I add a hostPort to the Pod the egress rule isn't applied and I get 'No route to host'. Calico deployed/configured with canal.yaml.

## Expected Behavior

egress rules should be applied to pod allowing access to url

## Current Behavior

egress rule is not applied and url access fails with 'No route to host'

## Steps to Reproduce (for bugs)

```
kubectl create namespace eg  || true
cat <<EOF | kubectl -n eg apply -f -
kind: NetworkPolicy
apiVersion: networking.k8s.io/v1
metadata:
  name: imds-egress
  labels:
    release: oac
    heritage: Tiller
spec:
  podSelector:
    matchLabels:
      netpol.koala.oracle.com/imds-egress: "true"
  policyTypes:
    - Egress
  egress:
    - to:
        - ipBlock:
            cidr: 169.254.169.254/32
      ports:
        - port: 80
          protocol: TCP
EOF
cat <<EOF | kubectl -n eg apply -f -
apiVersion: v1
kind: Pod
metadata:
  name: nc
  labels:
    netpol.koala.oracle.com/imds-egress: "true"
spec:
  containers:
  - name: nc
    image: container-registry.oracle.com/os/oraclelinux:7
    command: ["/bin/sh"]
    args: ["-c", "sleep 1d"]     
    ports:
    - containerPort: 9999
      protocol: TCP
---
apiVersion: v1
kind: Pod
metadata:
  name: nc-hostport
  labels:
    netpol.koala.oracle.com/imds-egress: "true"
spec:
  containers:
  - name: nc-hostport
    image: container-registry.oracle.com/os/oraclelinux:7
    command: ["/bin/sh"]
    args: ["-c", "sleep 1d"]     
    ports:
    - containerPort: 9999
      protocol: TCP
      hostPort: 9999
EOF

# works
kubectl -n eg exec -it nc -- bash -c 'curl -H "Authorization: Bearer Oracle" -L http://169.254.169.254/opc/v2/instance/'
# does not work, fails with 'curl: (7) Failed connect to 169.254.169.254:80; No route to host'
kubectl -n eg exec -it nc-hostport -- bash -c 'curl -H "Authorization: Bearer Oracle" -L http://169.254.169.254/opc/v2/instance/'      

```
## Context

Unable to activate network policy to secure our cluster as daemonsets require hostPort for optimized routing.

## Your Environment

* Calico version : 3.21.0
* Orchestrator version (e.g. kubernetes, mesos, rkt): oke 1.20.8
* Container runtime: cri-o://1.20.2
* Operating System and version: Oracle Linux Server 7.9, 5.4.17-2102.204.4.4.el7uek.x86_64
* Link to your project (optional): n/a

**Comment by johnsler** [comment]
Please note that this works in 3.19.3 (which we have switched to for the time being).

**Comment by lwr20** [comment]
I think this was fixed in Calico v3.21.1:
https://docs.projectcalico.org/release-notes/
> Fix bug where invalid port structures were being sent to Felix, preventing pods with hostPorts specified from working. libcalico-go #1545 (@caseydavenport)

**Comment by johnsler** [comment]
> I think this was fixed in Calico v3.21.1: https://docs.projectcalico.org/release-notes/
> 
> > Fix bug where invalid port structures were being sent to Felix, preventing pods with hostPorts specified from working. libcalico-go #1545 (@caseydavenport)

Confirming fixed in v3.21.1 - many thanks.
